### DISCUSSION TITLE
Device Type Groups

### DISCUSSION THREAD

**Discussion by byonas** [netbox-community/netbox#8404] [Ideas]
Grouping differnt models of the same series. For example, C3850-24P, C3850-24U, C3850-24T....etc to be on the same group. It would help in EOS/EOL.

**Comment by mtinberg** [comment]
Could you just create a custom field for this on the device type model, then you can query and sort device type records by model family.  You'll have to populate this field when creating a device by script or have a cron job come around and populate the field based on the device type or part number itself.  You can make it a required field or make a defined list of model families rather than a free-form text if you like and that should solve this issue without needing to change the base model with every field that everyone could possibly need, the base model should probably have the _minimum_ number of fields to be useful and rely on sites creating custom fields for their particular needs.


{
    "id": 44,
    "url": "https://netbox-qa/api/extras/custom-fields/44/",
    "display": "Device Model Series",
    "content_types": [
        "dcim.device",
        "dcim.devicetype"
    ],
    "type": {
        "value": "text",
        "label": "Text"
    },
    "name": "device_family",
    "label": "Device Model Series",
    "description": "Model family of device eg. Cat3850, N7K",
    "required": false,
    "filter_logic": {
        "value": "loose",
        "label": "Loose"
    },
    "default": null,
    "weight": 100,
    "validation_minimum": null,
    "validation_maximum": null,
    "validation_regex": "",
    "choices": []
}



—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: byonas ***@***.***>
Sent: Thursday, January 20, 2022 7:19 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Device Type Groups (Discussion #8404)


Grouping differnt models of the same series. For example, C3850-24P, C3850-24U, C3850-24T....etc to be on the same group. It would help in EOS/EOL.

—
Reply to this email directly, view it on GitHub<https://github.com/netbox-community/netbox/discussions/8404>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAS7UMYRLHE55KGQIFS5HR3UXADWXANCNFSM5MMXKPXQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
